xgi/generators/random.py

```diff
@@ -88,18 +88,16 @@ def fast_random_hypergraph(n, ps, order=None, seed=None):
         if p == 1:
             H.add_edges_from([e for e in combinations(nodes, d + 1)])
         elif p > 0:
-            r = random.random()
-            index = floor(log(r) / log(1 - p)) - 1  # -1 b/c zero indexing
-            max_index = comb(n, d + 1, exact=True)
+            index = geometric(p) - 1  # -1 b/c zero indexing
+            max_index = comb(n, d + 1, exact=True) - 1
```

Comment:
it's the -1 on this line that fixes the bug right?

Reply:
Correct. This is in line with PR #653.